airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -256,6 +266,85 @@ public static ServerRunnable getServer(final ServerFactory apiFactory, final Con
+            configRepository.writeStandardSourceDefinition(updatedDef);
+            LOGGER.info(
+                "migrateAllDefinitionsToContainSpec - Spec for Source Definition {} was successfully written to the db record.",
+                sourceDef.getName());
+          } else {
+            LOGGER.info(
+                "migrateAllDefinitionsToContainSpec - Failed to retrieve spec for Source Definition {}.",
+                sourceDef.getName());
+            throw new RuntimeException(String.format("Failed to retrieve spec for Source Definition %s", sourceDef.getName()));
+          }
+        }
+      } catch (final Exception e) {
+        trackSpecBackfillFailure(sourceDef.getDockerRepository(), sourceDef.getDockerImageTag(), e);
+      }
+    }
+
+    for (final StandardDestinationDefinition destDef : configRepository.listStandardDestinationDefinitions()) {
+      try {
+        if (destDef.getSpec() == null) {
+          LOGGER.info(
+              "migrateAllDefinitionsToContainSpec - Destination Definition {} does not have a spec. Attempting to retrieve spec...",
+              destDef.getName());
+          final SynchronousResponse<ConnectorSpecification> getSpecJob = schedulerClient
+              .createGetSpecJob(destDef.getDockerRepository() + ":" + destDef.getDockerImageTag());
+          if (getSpecJob.isSuccess()) {
+            LOGGER.info(
+                "migrateAllDefinitionsToContainSpec - Spec for Destination Definition {} was successfully retrieved. Writing to the db...",
+                destDef.getName());
+            final StandardDestinationDefinition updatedDef = Jsons.clone(destDef).withSpec(getSpecJob.getOutput());
+            configRepository.writeStandardDestinationDefinition(updatedDef);
+            LOGGER.info(
+                "migrateAllDefinitionsToContainSpec - Spec for Destination Definition {} was successfully written to the db record.",
+                destDef.getName());
+          } else {
+            LOGGER.info(
+                "migrateAllDefinitionsToContainSpec - Failed to retrieve spec for Destination Definition {}.",
+                destDef.getName());
+            throw new RuntimeException(String.format("Failed to retrieve spec for Destination Definition %s", destDef.getName()));
+          }
+        }
+      } catch (final Exception e) {
+        trackSpecBackfillFailure(destDef.getDockerRepository(), destDef.getDockerImageTag(), e);
+      }
+    }
+  }
+
+  private static void trackSpecBackfillFailure(final String dockerRepo, final String dockerImageTag, final Exception exception) {
+    final TrackingClient trackingClient = TrackingClientSingleton.get();
+    final ImmutableMap<String, Object> metadata = ImmutableMap.of(
+        "docker_image_name", dockerRepo,
+        "docker_image_tag", dockerImageTag,
+        "exception", exception);
+    trackingClient.track(UUID.fromString("000000"), "failed_spec_backfill", metadata);
```

Comment:
please add a comment explaining why we are going with this uuid. otherwise the next person who reads it will come ask you 😛

Reply:
I actually had to get rid of this and use a real workspace ID, because the tracking client yelled at me if I tried to pass in 0s.